Добавление второго задания диплома

Vrazhnov opened the pull request:
Добавлена вся работа по 2 проекту диплома:
Тесты по созданию пользователя
Тесты по логину пользователя
Тесты по изменению пользователя
Тесты по созданию заказа
Тесты по получению информации о заказе
Библиотеки:
allure.version 2.15.0
junit 4.13.2
rest-assured 5.3.0
gson 2.10.1
maven 2.22.2

Commit:
Добавление второго задания диплома

nikolay-977 commented on src/test/java/praktikum/order/OrderCreationTest.java:
```diff
@@ -0,0 +1,100 @@
+import praktikum.user.UserData;
+import praktikum.user.UserService;
+
+import java.util.Collections;
```
Необходимо исправить: Используй сочетание клавиш Ctrl+Alt+O (Option+Command+O для Mac) для оптимизации импортов (идея подсвечивает серым).Также можно установить курсор на корневую директорию и нажать клавиш Ctrl+Alt+L, отметить чек-боксы: cleanup code, а также optimize imports(для оптимизации импортов) и rearrange entries(для изменения порядка переменных и методов в соответствии с best practics)

nikolay-977 commented on src/test/java/praktikum/user/ChangeUserDataTest.java:
```diff
@@ -0,0 +1,58 @@
+                .assertThat()
+                .body("success", equalTo(true))
+                .and()
+                .statusCode(200);
```
Необходимо исправить порядок проверки - сначала проверяем statusCode, затем остальное. Иначе в случае если сервер недоступен мы получим NullPointerException, а не status сode 500-й.

nikolay-977 commented on src/test/java/praktikum/order/OrderCreationTest.java:
```diff
@@ -0,0 +1,100 @@
+    private UserData userData;
+    private UserService userService;
+    private Response response;
+    String accessToken;
```
Можно улучшить: Для всех полей следует указывать модификаторы доступа.

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из фалйа README.md). Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

Commit:
Исправления замечаний:
1) добавлено описание в README.md
2) выполнена оптимизация (cleanup code, optimize imports и rearrange entries)
3) исправлен порядок проверки в тесте - сначала проверяется statusCode, затем тело
4) добавлен недостающий модификатор доступа для accessToken

Commit:
в предыдущем коммите удалились файлы отчета - вернул их
